application-migrator-migrators/application-migrator-migrators-property/application-migrator-migrators-property-default/src/main/java/org/xwiki/contrib/migrator/migrators/internal/PropertyMigrationExecutor.java

```diff
@@ -0,0 +1,251 @@
+        }
+
+        return MigrationStatus.SUCCESS;
+    }
+
+    private void populizePropertyConversionChecker()
+    {
+        // There are examples of how the conversion check should be initialized
+        // In the future, we will support more types, and they can be defined here
+        logger.info("Initializing Checkers for property conversion ...");
+
+        PropertyConversionChecker pStringChecker = new PropertyConversionChecker(STRING_TYPE);
+        pStringChecker.getAllSupportedProperties().add(STRING_TYPE);
+
+        PropertyConversionChecker pTextAreaChecker = new PropertyConversionChecker(TEXT_AREA_TYPE);
+        pTextAreaChecker.getAllSupportedProperties().add(STRING_TYPE);
+
+        PropertyConversionChecker pNumberChecker = new PropertyConversionChecker(NUMBER_TYPE);
+        pNumberChecker.getAllSupportedProperties().add(STRING_TYPE);
+
+        PropertyConversionChecker pEMailChecker = new PropertyConversionChecker(EMAIL_TYPE);
+        pEMailChecker.getAllSupportedProperties().add(STRING_TYPE);
+
+        provider.getList().add(pStringChecker);
+        provider.getList().add(pTextAreaChecker);
+        provider.getList().add(pNumberChecker);
+        provider.getList().add(pEMailChecker);
+    }
+
+    private boolean areTwoPropertiesCompatible(PropertyClass oldPropertyType, PropertyClass newPropertyType)
+        throws MigrationException
+    {
+        logger.info("checking compatibility for property conversion ...");
+
+        PropertyConversionChecker localChecker = null;
+
+        // case 1 : same type
+        if (oldPropertyType.getName().equals(newPropertyType.getName())) {
+            return true;
+        }
+        // case 2 : different types
+        // get the proper checker
+        for (PropertyConversionChecker checker : provider.getList()) {
+            if (oldPropertyType.getName().equals(checker.getPropertyType())) {
+                localChecker = checker;
+            }
+        }
+        // check if conversion between the target type and the resource type is supported
+        try {
+            for (String supportedType : localChecker.getAllSupportedProperties()) {
+                if (newPropertyType.getName() == supportedType) {
+                    return true;
+                }
+            }
+        } catch (NullPointerException e) {
+            throw new MigrationException(
+                String.format("old Property type:  [%s] is not supported.", oldPropertyType.getName()), e);
+        }
+
+        return false;
+    }
+
+    private void migrateProperty(XWikiDocument document) throws MigrationException
+    {
+        logger.info("Migrating a property from type [{}] to new type [{}].", oldPropertyType.getName(),
+            newPropertyType.getName());
+
+        // set the new field with the old property class
+        document.setProperty(classReference, newPropertyType.getName(),
+            oldPropertyType.fromString(migrationParameters.getOldProperty()));
+    }
```

Comment:
Once the migration of the properties and the metadata of a document is done, you actually need to save the document itself, so that the changes that you did are persisted in the database. You should have `XWiki#saveDocument()` for that.